src/connector/src/sink/kinesis.rs

```diff
@@ -195,7 +195,7 @@ impl FormattedSink for KinesisSinkPayloadWriter {
     type V = Vec<u8>;
 
     async fn write_one(&mut self, k: Option<Self::K>, v: Option<Self::V>) -> Result<()> {
-        self.put_record(&k.unwrap(), v.unwrap_or_default())
+        self.put_record(&k.unwrap_or_default(), v.unwrap_or_default())
```

Comment:
Should we directly throw an error / panic here if k is None just like what we did before? Since according to kinesis, empty partition key will be accepted but also may lead to potential hot shard issue 🤔️.

Reply:
> Type: String
>
> Length Constraints: Minimum length of 1. Maximum length of 256.
>
> Required: Yes

Actually empty is not legal either... Guess I will at least throw an error.